File: nhd2-exp/data/skin/nmp/mainsetup.lua
-- mainsetup
local selected = 0

-- main
function main()
	local ret = neutrino.RETURN_REPAINT
	
	local m = neutrino.CMenuWidget(neutrino.LOCALE_MAINMENU_SETTINGS, neutrino.NEUTRINO_ICON_SETTINGS, 600)
	m:enablePaintFootInfo()
	m:enableShrinkMenu()

	item1 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_VIDEO, true, "", neutrino.CVideoSettings(), "")
	item1:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_VIDEOSETTINGS)
	item1:setHint(neutrino.LOCALE_MAINSETTINGS_VIDEO)
	item1:setIconName(neutrino.NEUTRINO_ICON_BUTTON_RED)

	item2 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_AUDIO)
	item2:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_AUDIOSETTINGS)
	item2:setActionKey(neutrino.CAudioSettings(), "")
	item2:setHint(neutrino.LOCALE_MAINSETTINGS_AUDIO)
	item2:setIconName(neutrino.NEUTRINO_ICON_BUTTON_GREEN)

	item3 = neutrino.CMenuForwarder(neutrino.LOCALE_PARENTALLOCK_PARENTALLOCK)
	item3:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_PARENTALLOCKSETTINGS)
	item3:setActionKey(neutrino.CParentalLockSettings(), "")
	item3:setHint(neutrino.LOCALE_PARENTALLOCK_PARENTALLOCK)
	item3:setIconName(neutrino.NEUTRINO_ICON_BUTTON_YELLOW)

	item4 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_NETWORK)
	item4:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_NETWORKSETTINGS)
	item4:setActionKey(neutrino.CNetworkSettings_getInstance(), "")
	item4:setHint(neutrino.LOCALE_MAINSETTINGS_NETWORK)
	item4:setIconName(neutrino.NEUTRINO_ICON_BUTTON_BLUE)

	item5 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_RECORDING)
	item5:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_RECORDINGSETTINGS)
	item5:setActionKey(neutrino.CRecordingSettings(), "");
	item5:setHint(neutrino.LOCALE_MAINSETTINGS_RECORDING)
	item5:setDirectKey(neutrino.RC_1)

	item6 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_STREAMING)
	item6:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_MOVIEPLAYERSETTINGS)
	item6:setActionKey(neutrino.CMoviePlayerSettings(), "");
	item6:setHint(neutrino.LOCALE_MAINSETTINGS_STREAMING)
	item6:setDirectKey(neutrino.RC_2)
	
	item7 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_OSD)
	item7:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_OSDSETTINGS)
	item7:setActionKey(None, "osdsetup");
	item7:setHint(neutrino.LOCALE_MAINSETTINGS_STREAMING)
	item7:setDirectKey(neutrino.RC_3)
	
	item8 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_LCD)
	item8:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_LCDSETTINGS)
	item8:setActionKey(neutrino.CLCDSettings(), "");
	item8:setHint(neutrino.LOCALE_MAINSETTINGS_LCD)
	item8:setDirectKey(neutrino.RC_4)
	
	item9 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_KEYBINDING)
	item9:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_REMOTECONTROLSETTINGS)
	item9:setActionKey(neutrino.CRemoteControlSettings(), "");
	item9:setHint(neutrino.LOCALE_MAINSETTINGS_KEYBINDING)
	item9:setDirectKey(neutrino.RC_5)
	
	item10 = neutrino.CMenuForwarder(neutrino.LOCALE_AUDIOPLAYERSETTINGS_GENERAL)
	item10:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_AUDIOPLAYERSETTINGS)
	item10:setActionKey(neutrino.CAudioPlayerSettings(), "");
	item10:setHint(neutrino.LOCALE_AUDIOPLAYERSETTINGS_GENERAL)
	item10:setDirectKey(neutrino.RC_6)
	
	item11 = neutrino.CMenuForwarder(neutrino.LOCALE_PICTUREVIEWERSETTINGS_GENERAL)
	item11:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_PICTUREVIEWERSETTINGS)
	item11:setActionKey(neutrino.CPictureViewerSettings(), "");
	item11:setHint(neutrino.LOCALE_PICTUREVIEWERSETTINGS_GENERAL)
	item11:setDirectKey(neutrino.RC_7)
	
	item12 = neutrino.CMenuForwarder(neutrino.LOCALE_MAINSETTINGS_MISC)
	item12:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_MISCSETTINGS)
	item12:setActionKey(None, "miscsetup");
	item12:setHint(neutrino.LOCALE_MAINSETTINGS_MISC)
	item12:setDirectKey(neutrino.RC_8)
	
	item13 = neutrino.CMenuForwarder(neutrino.LOCALE_HDD_SETTINGS)
	item13:setItemIcon(neutrino.NEUTRINO_ICON_MENUITEM_HDDSETTINGS)
	item13:setActionKey(neutrino.CHDDMenuHandler(), "");
	item13:setHint(neutrino.LOCALE_HDD_SETTINGS)
	item13:setDirectKey(neutrino.RC_9)

	m:addItem(item1)
	m:addItem(item2)
	m:addItem(item3)
	m:addItem(item4)
	m:addItem(item5)
	m:addItem(item6)
	m:addItem(item7)
	m:addItem(item8)
	m:addItem(item9)
	m:addItem(item10)
	m:addItem(item11)
	m:addItem(item12)
	m:addItem(item13)
	
	if selected < 0 then
		selected = 0
	end

	m:setSelected(selected)

	ret = m:exec(None, "")
	
	selected = m:getSelected()
	 
	local actionKey = m:getActionKey()
	
	if actionKey == "miscsetup" then
		neutrino.CNeutrinoApp_getInstance():startSkin("miscsetup")
	end
	if actionKey == "osdsetup" then
		neutrino.CNeutrinoApp_getInstance():startSkin("osdsetup")
	end
	
	if m:getExitPressed() ~= true and ret == neutrino.RETURN_REPAINT then
		main()
	end

	return ret
end

main()

File: nhd2-exp/src/gui/main_setup.cpp
/*
	Neutrino-GUI  -   DBoxII-Project

	$id: main_setup.cpp 2015.12.22 21:31:30 mohousch $
	
	Copyright (C) 2001 Steffen Hehn 'McClean'
	and some other guys
	Homepage: http://dbox.cyberphoria.org/

	License: GPL

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation; either version 2 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
*/ 

#ifdef HAVE_CONFIG_H
#include <config.h>
#endif

#include <global.h>
#include <neutrino.h>

#include <stdio.h>

#include <gui/main_setup.h>

#include <gui/audio_setup.h>
#include <gui/video_setup.h>
#include <gui/parentallock_setup.h>
#include <gui/network_setup.h>
#include <gui/movieplayer_setup.h>
#include <gui/osd_setup.h>
#include <gui/audioplayer_setup.h>
#include <gui/pictureviewer_setup.h>
#include <gui/lcd_setup.h>
#include <gui/rc_setup.h>
#include <gui/recording_setup.h>
#include <gui/misc_setup.h>
#include <gui/hdd_menu.h>

#include <system/debug.h>
#include <system/setting_helpers.h>
#include <system/helpers.h>


// main settings
int CMainSettingsMenu::exec(CMenuTarget* parent, const std::string& actionKey)
{
	dprintf(DEBUG_NORMAL, "CMainSettingsMenu::exec: actionKey: %s\n", actionKey.c_str());
	
	int ret = RETURN_REPAINT;
	
	if(parent)
		parent->hide();
	
	showMenu();
	
	return ret;
}

// showmenu
void CMainSettingsMenu::showMenu(void)
{
	dprintf(DEBUG_NORMAL, "CMainSettingsMenu::showMenu:\n");
	
	int shortcutMainSettings = 1;

	CMenuWidget * mainSettings = new CMenuWidget(_("Main Settings"), NEUTRINO_ICON_SETTINGS);

	mainSettings->setWidgetMode(MODE_MENU);
	mainSettings->setWidgetType(WIDGET_TYPE_CLASSIC);
	mainSettings->enableShrinkMenu();
	mainSettings->setMenuPosition(MENU_POSITION_LEFT);
	mainSettings->enablePaintFootInfo();
	mainSettings->enablePaintDate();

	// video settings
	mainSettings->addItem(new CMenuForwarder(_("Video settings"), true, NULL, new CVideoSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL, NEUTRINO_ICON_MENUITEM_VIDEOSETTINGS));

	// audio settings
	mainSettings->addItem(new CMenuForwarder(_("Audio settings"), true, NULL, new CAudioSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL, NEUTRINO_ICON_MENUITEM_AUDIOSETTINGS));

	// parentallock
	mainSettings->addItem(new CMenuForwarder(_("Parentallock settings"), true, NULL, new CParentalLockSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL,  NEUTRINO_ICON_MENUITEM_PARENTALLOCKSETTINGS));

	// network settings
	mainSettings->addItem(new CMenuForwarder(_("Network settings"), true, NULL,  CNetworkSettings::getInstance(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL, NEUTRINO_ICON_MENUITEM_NETWORKSETTINGS));

	// recording settings
	mainSettings->addItem(new CMenuForwarder(_("Recording settings"), true, NULL, new CRecordingSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL,  NEUTRINO_ICON_MENUITEM_RECORDINGSETTINGS));

	// movieplayer settings
	mainSettings->addItem(new CMenuForwarder(_("Movie Player settings"), true, NULL, new CMoviePlayerSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL,  NEUTRINO_ICON_MENUITEM_MOVIEPLAYERSETTINGS));

	//OSD settings
	mainSettings->addItem(new CMenuForwarder(_("OSD settings"), true, NULL, new COSDSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL, NEUTRINO_ICON_MENUITEM_OSDSETTINGS));

	// vfd/lcd settings
	//if(CVFD::getInstance()->has_lcd)
		mainSettings->addItem(new CMenuForwarder(_("LCD settings"), true, NULL, new CLCDSettings(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL,  NEUTRINO_ICON_MENUITEM_LCDSETTINGS));	

	// remote control settings
	mainSettings->addItem(new CMenuForwarder(_("Remote Control settings"), true, NULL, new CRemoteControlSettings(), NULL, RC_red, NEUTRINO_ICON_BUTTON_RED, NEUTRINO_ICON_MENUITEM_REMOTECONTROLSETTINGS));

	// audioplayer settings
	mainSettings->addItem(new CMenuForwarder(_("Audio Player settings"), true, NULL, new CAudioPlayerSettings(), NULL, RC_green, NEUTRINO_ICON_BUTTON_GREEN, NEUTRINO_ICON_MENUITEM_AUDIOPLAYERSETTINGS));
	
	// pictureviewer settings
	mainSettings->addItem(new CMenuForwarder("Picture Viewer settings", true, NULL, new CPictureViewerSettings(), NULL, RC_yellow, NEUTRINO_ICON_BUTTON_YELLOW, NEUTRINO_ICON_MENUITEM_PICTUREVIEWERSETTINGS));

	// misc settings
	mainSettings->addItem(new CMenuForwarder(_("Misc settings"), true, NULL, new CMiscSettingsMenu(), NULL, RC_blue, NEUTRINO_ICON_BUTTON_BLUE, NEUTRINO_ICON_MENUITEM_MISCSETTINGS));

	//HDD settings
	mainSettings->addItem(new CMenuForwarder(_("HDD settings"), true, NULL, new CHDDMenuHandler(), NULL, CRCInput::convertDigitToKey(shortcutMainSettings++), NULL, NEUTRINO_ICON_MENUITEM_HDDSETTINGS));
	
	mainSettings->exec(NULL, "");
	mainSettings->hide();
	delete mainSettings;
	mainSettings = NULL;
}

